setup.py

```diff
@@ -29,6 +29,7 @@
     long_description = (Path(__file__).parent / "README.md").read_text(),
     long_description_content_type="text/markdown",
     license='Custom',
+    python_requires='>=3.7, <3.11',
```

Comment:
Should we limit this here? Using python >3.10 should be fine unless you use tensorflow 2.x, right?